src/web/mjs/connectors/WebComicsApp.mjs

```diff
@@ -1,74 +1,156 @@
 import Connector from '../engine/Connector.mjs';
+import Manga from '../engine/Manga.mjs';
 
-/**
- *
- */
 export default class WebComicsApp extends Connector {
 
-    /**
-     *
-     */
     constructor() {
         super();
         super.id = 'webcomicsapp';
         super.label = 'WebComicsApp';
         this.tags = [ 'webtoon', 'english' ];
-        this.url = 'http://www.webcomicsapp.com';
+        this.url = 'https://www.webcomicsapp.com';
+
+        this.queryMangaTitleFromURI = 'div.book-info div.card-right h5';
+        this.queryMangas = 'section.mangas div.row div.col-md-3 a';
+        this.queryChapters = 'div.chapter-list-pc ul.chapter-contents li[class*="ch_"]';
+        this.queryPages = 'div.read-container div.read-box source';
+        this.requestOptions.headers.set('x-referer', `${this.url}/`);
+        this.titleRegex = /[\s|\p{P}]+/gu;
+        this.token = null;
+    }
+
+    async _initializeConnector() {
+        let script = `
+            new Promise((resolve, reject) => {
+                setTimeout(() => {
+                    try {
+                        resolve(!!localStorage.tk ? localStorage.tk : null);
+                    } catch(error) {
+                        reject(error);
+                    }
+                }, 2500);
+            });
+        `;
+        const uri = new URL(this.url);
+        const request = new Request(uri, this.requestOptions);
+        this.token = await Engine.Request.fetchUI(request, script);
+    }
+
+    async _getMangaFromURI(uri) {
+        const request = new Request(new URL(uri), this.requestOptions);
+        const data = await this.fetchDOM(request, this.queryMangaTitleFromURI);
+        const id = uri.pathname;
+        const title = data[0].textContent.trim();
+        return new Manga(this, id, title);
+    }
+
+    async _getMangas() {
+        const uri = new URL('/genres.html?category=all', this.url);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, this.queryMangas);
+        return data.map(element => {
+            const mangaTitle = element.querySelector('h5').textContent.trim();
+            return {
+                id: `${this.url}/comic/${mangaTitle.replace(this.titleRegex, '-')}/${this.getRelativeLink(element).match(/=(\w+)$/)[1]}`,
```

Comment:
Not necessary to prefix with `this.url`, this is already added in `_getChaptersWithLogin`

Reply:
Thanks for mentioning! Forgot about this, fixed.